### ISSUE TITLE
Can you give more concrete input layer examples??

### ISSUE THREAD

**Issue by kagecom** [PKU-ICST-MIPL/QaDWH_TMM2018#2]
I read the "data/flickr25k" folder, but i still don't know how to organize the input.

According to your train prototxt, file, there are two input layers,  the source of the first one input layer is  
"/home/junchao/hashing/data/flickr25k/flickr25k_triplet_rs15w_noconflict_sf.txt", where some lines are 

/home/zhangjian/workspace/dataset/MIRFlickr/resized/im22324.jpg 3149
/home/zhangjian/workspace/dataset/MIRFlickr/resized/im261.jpg 890
/home/zhangjian/workspace/dataset/MIRFlickr/resized/im13731.jpg 1826

what does thess number 3149, 890, 1826 mean?

And the source of the second input layer is "/home/junchao/hashing/data/flickr25k/hdf5/flickr25k_triplet_rs15w_noconflict_sf_multi_ label_h5.list"

what does this input file contain?

Please give some examples, thank you.

**Comment by PKU-ICST-MIPL** [comment]
In flickr25k_triplet_rs15w_noconflict_sf.txt, the images are triplet pairs, such as (1,2,3) (4,5,6) (7,8,9), which the number implies the line in the txt. For pair (1,2,3), 1 and 2 has common label, while 1 and 3 has no common label.
The numbers 3149,890,1826 are meaningless, which can be ignored. The number is used to fit the input format of the caffe. 
For more details, see the code of the triplet loss layer.

The flickr25k_triplet_rs15w_noconflict_sf_multi_ label_h5.list is list of h5 file path. The h5 file contains multi label for the images, which is denoted as a vector, for example, (0,0,0,1,1,0,0,0,0,0,0,0,1,0 ...).  For more details, see the code of the multi_label layer.

**Comment by liyakun** [comment]
@PKU-ICST-MIPL thanks for providing the source code. Just for clarity, 
 `flickr25k_triplet_rs15w_noconflict_sf_multi_ label_h5.list` contains the list of h5 files, and the h5 files containing the encoded label vectors, right? Thanks.

**Comment by PKU-ICST-MIPL** [comment]
> @PKU-ICST-MIPL thanks for providing the source code. Just for clarity,
> `flickr25k_triplet_rs15w_noconflict_sf_multi_ label_h5.list` contains the list of h5 files, and the h5 files containing the encoded label vectors, right? Thanks.

Yes, thanks for your suggestion, we have modified the answer to be more clear.

**Comment by Xinguo806** [comment]
flickr25k_triplet_rs15w_noconflict_sf_multi_ label_h5.list is missing in the current version, could you kindly provide the source file?

**Comment by PKU-ICST-MIPL** [comment]
> flickr25k_triplet_rs15w_noconflict_sf_multi_ label_h5.list is missing in the current version, could you kindly provide the source file?

The flickr25k_triplet_rs15w_noconflict_sf_multi_ label_h5.list is list of h5 file path. The h5 file contains multi label for the images, which is denoted as a vector, for example, (0,0,0,1,1,0,0,0,0,0,0,0,1,0 ...). For more details, see the code of the multi_label layer.
you should conduct the h5 files for multi-labels yourself, and write the file path in flickr25k_triplet_rs15w_noconflict_sf_multi_ label_h5.list.

**Comment by Xinguo806** [comment]
Okay, I see, thanks for your reply.

**Comment by wzb1005** [comment]
Hi,
(Some findings of "mirflickr25K" dataset)
I find that in the "common_tags.txt"  file of "mirflickr25K" dataset, there are 1384 classes, and in the "mirflickr25k_annotation“ directory, there are 40 classes, How does the 38 classes got?? What the name of the  classes?

**Comment by PKU-ICST-MIPL** [comment]
> Hi,
> (Some findings of "mirflickr25K" dataset)
> I find that in the "common_tags.txt" file of "mirflickr25K" dataset, there are 1384 classes, and in the "mirflickr25k_annotation“ directory, there are 40 classes, How does the 38 classes got?? What the name of the classes?

Please refer to the http://press.liacs.nl/mirflickr/mirdownload.html. 
the "mirflickr25k_annotations_v080.zip" contains 38 classes of the image. 
the tag is regarded as text in the paper.

**Comment by iTomxy** [comment]
Is it the order of labels in the h5 file should be one-to-one correspondent to that in *flickr25k_triplet_rs15w_noconflict_sf.txt* and *flickr25k_qtest_resize.txt*?

**Comment by PKU-ICST-MIPL** [comment]
> Is it the order of labels in the h5 file should be one-to-one correspondent to that in _flickr25k_triplet_rs15w_noconflict_sf.txt_ and _flickr25k_qtest_resize.txt_?

Yes, the caffe will load the data and labels with the order of these files.
